src/sdam/server_selection.ts

```diff
@@ -113,7 +145,7 @@ function maxStalenessReducer(
       primaryFilter
     )[0];
 
```

Comment:
There are a chunk of "clean ups" that I put in this PR.  I've separated them into a single commit to make reviewing easier, but basically they boil down to just replacing `Array.reduce` calls that perform as filters with `filter`.  You can view those changes by just reviewing the first commit of this PR.